compiler/rustc_session/src/config.rs

```diff
@@ -1418,102 +1440,83 @@ impl CheckCfg {
             ExpectedValues::Some(values)
         };
 
-        // NOTE: This should be kept in sync with `default_configuration`
+        macro_rules! ins {
+            ($name:expr, $values:expr) => {
+                self.expecteds.entry($name).or_insert_with($values)
+            };
+        }
 
-        let panic_values = &PanicStrategy::all();
+        // Symbols are inserted in alphabetical order as much as possible.
+        // The exceptions are where control flow forces things out of order.
+        //
+        // NOTE: This should be kept in sync with `default_configuration`.
+        // Though symbols inserted conditionally in `default_configuration` are
+        // inserted unconditionally here.
 
-        let atomic_values = &[
-            sym::ptr,
-            sym::integer(8usize),
-            sym::integer(16usize),
-            sym::integer(32usize),
-            sym::integer(64usize),
-            sym::integer(128usize),
-        ];
+        ins!(sym::debug_assertions, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::doc, no_values);
+        // njn: not in default_configuration!
+        ins!(sym::doctest, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::miri, no_values);
```

Comment:
`doc`, `doctest` and `miri` are set by rustdoc and miri respectively. They are never set by rustc, but should still not trigger a lint as `cargo doc`, `cargo test` and `cargo miri run` respectively can set them anyway.

Reply:
Yes exactly, when enabled, checking the configs happen every time for every configs.
When doing `cargo build` we don't want to lint on `rustdoc` configs and the inverse when doing `cargo doc` we don't want to lint on `rustc` configs. It would be annoying and make the feature completely useless.

As for `miri` config it's a bit tangent, Miri is a nightly-only tool, I could be convinced to not include it and have the user include it manually; but this should be discussed elsewhere I suppose.

Reply:
:+1: